### ISSUE TITLE
Peer communication issues over spotty connection freezes game

### ISSUE THREAD

**Issue by brianismeta** [brianismeta/Ortho1#12]
Peer communication issues over spotty connection freezes game

When playing with another player, either the internet connection went down or maybe its strength changed, but the game simply froze.  It appears webrtc had a connection issue.  The game does not recover well from this, but maybe it should be able to recover.

![Peer-connection-issues](https://user-images.githubusercontent.com/101431261/217615374-f80a0191-686f-49e2-821f-72e4829e2444.png)

May try to repro and see if there is a way to recover a broken connection.

**Comment by brianismeta** [comment]
I think this was due to UDP connection.  Sometimes you think or assume things that are not true.  PeerJS has a configuration option when connecting to a peer whether the connection should be ‘reliable’ (true or false).  Without any additional explanation of what it means.

🤔

As TURN servers can operate over a TCP or UDP connection, it was an assumption on my part that ‘reliable’ translated to TCP. 😕  I was fooled, because I could also see TCP connections in SysInternal’s TCPView tool, and when I ended them, it caused errors in the console.  However when I started looking closer, the packet counts were not increasing 😳.  Using wire shark, I found out that it was actually using UDP!

😨😨😨

To make the connections over TCP need to specify the TURN servers explicitly, with the ?transport=tcp at the end of the URL.  Thanks to this Medium article for explaining this: https://lnkd.in/g_ZgnkCi
